packages/client/src/generation/generateClient.ts

```diff
@@ -141,26 +164,38 @@ export async function buildClient({
   fileMap['default.js'] = await JS(defaultClient)
   fileMap['default.d.ts'] = await TS(defaultClient)
   fileMap['index-browser.js'] = await BrowserJS(nodeClient)
-  fileMap['package.json'] = JSON.stringify(pkgJson, null, 2)
   fileMap['edge.js'] = await JS(edgeClient)
   fileMap['edge.d.ts'] = await TS(edgeClient)
 
   if (generator.previewFeatures.includes('driverAdapters')) {
-    // in custom outputs, `index` shows a warning. if it is loaded, it means
-    // that the export map is not working for the user so we display them an
-    // `importWarning`. If the exports map works, `default` will be loaded.
-    if (generator.isCustomOutput === true) {
-      const nodeWarnTsClient = new TSClient({
-        ...nodeClientOptions,
-        reusedTs: 'default',
-        reusedJs: 'default',
-        importWarning: true,
-      })
-
-      fileMap['default.js'] = fileMap['index.js']
-      fileMap['default.d.ts'] = fileMap['index.d.ts']
-      fileMap['index.js'] = await JS(nodeWarnTsClient)
-      fileMap['index.d.ts'] = await TS(nodeWarnTsClient)
+    // The trampoline client points to #main-entry-point (see below).  We use
+    // imports similar to an exports map to ensure correct imports.❗ Before
+    // going GA, please notify @millsp as some things can be cleaned up:
+    // - defaultClient can be deleted since trampolineTsClient will replace it.
+    //   - Special handling of . paths in TSClient.ts can also be removed.
+    // - The main @prisma/client exports map can be simplified:
+    //   - Everything can point to `default.js`, including browser fields.
+    //   - Exports map's `.` entry can be made like the others (e.g. `./edge`).
+    // - exportsMapDefault can be deleted as it's only needed for defaultClient:
+    //   - #main-entry-point can handle all the heavy lifting on its own.
+    //   - Always using #main-entry-point is kept for GA (small breaking change).
+    //   - exportsMapDefault can be inlined down below and removed everywhere else.
+    // In short: A lot can be simplified, but can only happen in GA & P6.
+    fileMap['default.js'] = await JS(trampolineTsClient)
+    fileMap['default.d.ts'] = await TS(trampolineTsClient)
+    fileMap['wasm-worker-loader.js'] = `export default (await import('./query_engine_bg.wasm')).default`
+    fileMap['wasm-edge-light-loader.js'] = `export default (await import('./query_engine_bg.wasm?module')).default`
+    pkgJson['browser'] = 'default.js' // also point to the trampoline client otherwise it is picked up by cfw
+    pkgJson['imports'] = {
+      // when `import('#wasm-engine-loader')` is called, it will be resolved to the correct file
+      '#wasm-engine-loader': {
+        'edge-light': './wasm-edge-light-loader.js',
+        workerd: './wasm-worker-loader.js',
+        worker: './wasm-worker-loader.js',
+        default: './wasm-worker-loader.js',
+      },
```

Comment:
This fixes the DX for the engine loading on Vercel. By having a custom engine laoder for Vercel and Cloudflare we can remove the extra configuration needed in `next.config.js` to be able to load the wasm. Now it will load out of the box.